internal/experimental/registry/client.go

```diff
@@ -37,6 +37,13 @@ import (
 	"helm.sh/helm/v3/pkg/helmpath"
 )
 
+// See https://github.com/helm/helm/issues/10166
+const registryUnderscoreMessage = `
+OCI artifact references do not support the plus sign(+). To solve this Helm
+adopts the convention of changing plus (+) to underscore (_) in chart version
+references during registry push - and OCI artifact reference from underscore
+back to plus on registry pull.`
```

Comment:
Hey @mattfarina Yes I think noting semver in the message makes sense
- Updated that and added your co-authorship in the commit trailer
- Added newlines to message output instead of the message const itself